Code/Framework/AzCore/AzCore/std/utility/expected.h

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/std/typetraits/is_swappable.h>
+#include <AzCore/std/utility/expected_internal.h>
+
+namespace AZStd
+{
+    //! Used to pass wrap the error type supplied to the construct an expected object
```

Comment:
nit/update: `//! Used to pass wrap the error type supplied to the construct an expected object` -> `//! Wraps the error type supplied to construct an expected object`

Reply:
Also should this say 'unexpected' object?

Reply:
What makes the terminology confusing between "unexpected" and "error" is that the type alias is [error_type](https://en.cppreference.com/w/cpp/utility/expected) which has the description of "the type of the unexpected value"